managed/src/main/java/com/yugabyte/yw/models/Backup.java

```diff
@@ -190,14 +190,17 @@ public static Backup fetchByTaskUUID(UUID taskUUID) {
       .findOne();
   }
 
-  public static List<Backup> getExpiredBackups(UUID scheduleUUID) {
+  public static List<Backup> getExpiredBackups() {
     // Get current timestamp.
     Date now = new Date();
-    return Backup.find.query().where()
-      .eq("schedule_uuid", scheduleUUID)
+    List<Backup> backupList =  Backup.find.query().where()
       .lt("expiry", now)
       .eq("state", BackupState.Completed)
       .findList();
+    return backupList.stream().filter(
```

Comment:
Just a thought, why not fetch all universes first, make a set of the UUIDs and then try to see if the universe UUID for the backup exists in the set? The reason I think that would be better is since we wouldn't have to query the DB multiple times to get the universe. Eventually if the backups are a lot, things might pile up.

Reply:
Done